internal/impl/gcp/cache_cloud_storage.go

```diff
@@ -60,6 +70,25 @@ func newGcpCloudStorageCacheFromConfig(parsedConf *service.ParsedConfig) (*gcpCl
 	}, nil
 }
 
+func getClientOptionsForCloudStorageCache(parsedConf *service.ParsedConfig) ([]option.ClientOption, error) {
```

Comment:
Having a unique `getClientOptionsForXXX` seems a bit error prone as they all have to reproduce the credentials logic, could we collapse this into a single `getAuthClientOptions(parsedConf *service.ParsedConfig)` that they all share?

Reply:
@Jeffail added a new commit to update to consolidate the options logic into single function `getClientOptionWithCredential()` and call that from from all the files.